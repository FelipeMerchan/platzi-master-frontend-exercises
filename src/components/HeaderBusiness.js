import React from 'react'

import MenuBusiness from './MenuBusiness'
import './styles/HeaderBusiness.css'
import businessLogo from '../images/brand.JPG'

function HeaderBusiness(props) {
  return (
    <header className="HeaderBusiness">
        <figure className="HeaderBusiness__brandContainer">
          <img className="HeaderBusiness__brand" width="300" src={businessLogo} alt="Empresa.com" />
        </figure>
        <MenuBusiness onClick={props.onClick} />
    </header>
  )
}

export default HeaderBusiness