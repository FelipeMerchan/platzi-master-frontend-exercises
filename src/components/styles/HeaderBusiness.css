.HeaderBusiness {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: var(--space-2) 0;
  border-bottom: 1px solid #dfdfdf;
  font-family: var(--font-3);
}

.HeaderBusiness__brand {
  max-width: 250px;
}