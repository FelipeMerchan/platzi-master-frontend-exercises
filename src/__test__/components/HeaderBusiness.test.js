import React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'
import '../../setupTest'
import HeaderBusiness from '../../components/HeaderBusiness'
import brandImage from '../../images/brand.JPG'

describe('Probar el componente <HeaderBusiness />', () => {
  const headerBusiness = shallow(<HeaderBusiness />)
  test('Render del componente HeaderBusiness', () => {
    expect(headerBusiness.length).toEqual(1)
  })
  test('Render de el logotipo', () => {
    expect(headerBusiness.find('img').prop('src')).toEqual(brandImage)
  })
})

describe('HeaderBusiness Snapshot', () => {
  const headerBusiness = create(<HeaderBusiness />)
  test('Validar la UI del componente HeaderBusiness', () => {
    expect(headerBusiness.toJSON()).toMatchSnapshot()
  })
})