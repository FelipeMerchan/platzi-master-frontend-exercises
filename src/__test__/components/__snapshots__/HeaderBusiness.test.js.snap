// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBusiness Snapshot Validar la UI del componente HeaderBusiness 1`] = `
<header
  className="HeaderBusiness"
>
  <figure
    className="HeaderBusiness__brandContainer"
  >
    <img
      alt="Empresa.com"
      className="HeaderBusiness__brand"
      src="brand.JPG"
      width="300"
    />
  </figure>
  <nav
    className="MenuBusiness"
  >
    <ul>
      <li
        className="MenuBusiness__item"
      >
        <a
          data-button="nav"
          href="#home"
          id="home"
          onClick={[Function]}
          role="button"
        >
          Inicio
        </a>
      </li>
      <li
        className="MenuBusiness__item"
      >
        <a
          data-button="nav"
          href="#aboutUs"
          id="aboutUs"
          onClick={[Function]}
          role="button"
        >
          Acerca
        </a>
      </li>
      <li
        className="MenuBusiness__item"
      >
        <a
          data-button="nav"
          href="#services"
          id="services"
          onClick={[Function]}
          role="button"
        >
          Servicios
        </a>
      </li>
      <li
        className="MenuBusiness__item"
      >
        <a
          data-button="nav"
          href="#products"
          id="products"
          onClick={[Function]}
          role="button"
        >
          Productos
        </a>
      </li>
      <li
        className="MenuBusiness__item"
      >
        <a
          data-button="nav"
          href="#contact"
          id="contact"
          onClick={[Function]}
          role="button"
        >
          Contacto
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
